### PR TITLE
Feature/document tls servername

### PR DESCRIPTION
**Pull request by in0rdr**
Hi @schavis,

Thanks for your review on https://github.com/hashicorp/vault/pull/12982

I was unable to apply your suggestions because I deleted the fork with the local change previously.

In any case, find here the update with all the suggestions.

The suggestions are really helpful, so If you find anything else we could improve just let us know.

Thanks

### CODE REVIEW

File: website/content/docs/platform/k8s/helm/examples/ha-tls.mdx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+TLS](/docs/platform/k8s/helm/examples/standalone-tls) to create a TLS
+certificate.
+
+## Bootstrap your Raft cluster
+
+Without proper configuration, you will see the following warning before cluster
+initialization:
+```shell
+core: join attempt failed: error="error during raft bootstrap init call: Put "https://vault-${N}.${SERVICE}:8200/v1/sys/storage/raft/bootstrap/challenge": x509: certificate is valid for ${SERVICE}, ${SERVICE}.${NAMESPACE}, ${SERVICE}.${NAMESPACE}.svc, ${SERVICE}.${NAMESPACE}.svc.cluster.local, not vault-${N}.${SERVICE}"
+```
```
Comment (schavis) [inline website/content/docs/platform/k8s/helm/examples/ha-tls.mdx:35]: I'm not sure having this as an independent section is useful. What do you think of dropping the heading and making it the second paragraph in the introduction? That way you're outlining the situation in the first paragraph, showing the symptom in the second paragraph, then introducing your example solutions.

File: website/content/docs/platform/k8s/helm/examples/ha-tls.mdx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    leader_ca_cert_file = "/vault/tls/vault.ca"
+  }
+}
+```
+
+</CodeBlockConfig>
+
+## Solution 2:  Add a load balancer to your Raft configuration
+
+If you have a load balancer for your Vault cluster, you can add  a single
```
Comment (schavis) [inline website/content/docs/platform/k8s/helm/examples/ha-tls.mdx:81]:
```suggestion
If you have a load balancer for your Vault cluster, you can add a single
```

Remove stray space

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/docs/platform/k8s/helm/examples/ha-tls.mdx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+page_title: 'HA Cluster with TLS'
+sidebar_current: 'docs-platform-k8s-examples-ha-tls'
+description: |-
+  Describes how to set up a Raft HA Vault cluster with TLS certificate
+---
+
+# Raft HA server with TLS
+
+The overview for [Integrated Storage and
+TLS](/docs/concepts/integrated-storage#integrated-storage-and-tls) covers the
```
Comment (schavis) [inline website/content/docs/platform/k8s/helm/examples/ha-tls.mdx:12]:
```suggestion
TLS](/vault/docs/concepts/integrated-storage#integrated-storage-and-tls) covers the
```

File: website/content/docs/platform/k8s/helm/examples/ha-tls.mdx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+Raft cluster.
+
+The examples below demonstrate two specific solutions. Both solutions ensure
+that the common name (CN) used for the `leader_api_addr` in the Raft stanza
+matches the name(s) listed in the TLS certificate.
+
+## Before you start
+
+1. Follow the steps from the example [HA Vault Cluster with Integrated
+Storage](/docs/platform/k8s/helm/examples/ha-with-raft) to build the cluster.
```
Comment (schavis) [inline website/content/docs/platform/k8s/helm/examples/ha-tls.mdx:23]:
```suggestion
Storage](/vault/docs/platform/k8s/helm/examples/ha-with-raft) to build the cluster.
```

File: website/content/docs/platform/k8s/helm/examples/ha-tls.mdx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+2. Follow the examples and instructions in [Standalone Server with
+TLS](/docs/platform/k8s/helm/examples/standalone-tls) to create a TLS
```
Comment (schavis) [inline website/content/docs/platform/k8s/helm/examples/ha-tls.mdx:26]:
```suggestion
TLS](/vault/docs/platform/k8s/helm/examples/standalone-tls) to create a TLS
```

File: website/content/docs/platform/k8s/helm/examples/ha-tls.mdx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+initialization:
+```shell
+core: join attempt failed: error="error during raft bootstrap init call: Put "https://vault-${N}.${SERVICE}:8200/v1/sys/storage/raft/bootstrap/challenge": x509: certificate is valid for ${SERVICE}, ${SERVICE}.${NAMESPACE}, ${SERVICE}.${NAMESPACE}.svc, ${SERVICE}.${NAMESPACE}.svc.cluster.local, not vault-${N}.${SERVICE}"
+```
+
+## Solution 1: Use auto-join and set the TLS server in your Raft configuration
+
+The join warning disappears if you use auto-join and set the expected TLS
+server name (`${CN}`) with
+[`leader_tls_servername`](/docs/configuration/storage/raft#leader_tls_servername)
```
Comment (schavis) [inline website/content/docs/platform/k8s/helm/examples/ha-tls.mdx:41]:
```suggestion
[`leader_tls_servername`](/vault/docs/configuration/storage/raft#leader_tls_servername)
```

**Review by schavis — CHANGES_REQUESTED** [review]

**Comment by schavis** [comment]
> I was unable to apply your suggestions because I deleted the fork with the local change previously.

D'oh! Thanks for taking the time to resubmit the PR! I added some additional comments. Most are fixes for the links that I missed the first time through.

Also, I forgot to mention on the older PR that you'll need to also update the TOC file (/website/data/docs-nav-data.json) so the new doc shows up in the left-hand navigation. The Helm examples section is currently around ~line 1761

**Comment by in0rdr** [comment]
Thanks a lot for the review, I learn a lot about how these docs pages are being built.

Also, I have to admit that I like the example much better now, thanks for the suggestions :smile: 

I also decided to make the page title a bit more similar to the the existing documentation examples, mostly because we reference two examples in the text.

File: website/data/docs-nav-data.json [hunk 7]
```diff
@@ -1796,6 +1796,10 @@
                     "title": "HA Cluster with Raft",
                     "path": "platform/k8s/helm/examples/ha-with-raft"
                   },
```
Comment (in0rdr) [inline website/data/docs-nav-data.json:1798]: I decided to put the link in the sidebar right after the existing two examples

**Review by schavis — APPROVED** [review]

**Comment by schavis** [comment]
LGTM! Feel free to merge

**Comment by in0rdr** [comment]
Thanks @schavis I did a rebase but I'm not authorized to merge/deploy. Can you take care of this? Thank you.

**Comment by schavis** [comment]
> Thanks @schavis I did a rebase but I'm not authorized to merge/deploy. Can you take care of this? Thank you.

Apologies for the delay, I was out sick last week. I'll get this merged today.

**Pull request merged**